src/builder/builder.go

```diff
@@ -116,7 +116,7 @@ func (b *Builder) ReadBuilderConf() {
 		r := regexp.MustCompile(h.re)
 		for _, i := range lines {
 			if m := r.FindIndex([]byte(i)); m != nil {
-				*h.dest = i[m[1]:]
+				*h.dest = os.ExpandEnv(i[m[1]:])
```

Comment:
IMHO this should be validated in the case that someone uses a $VAR and it's not set, which ExpandEnv will return "" for, and will cause problems later on in the build paths. Can we check this and exit if something sane isn't returned?

Reply:
@tmarcu ExpandEnv is an API which replaces the environment variable with it's value. Although I already that if a variable which isn't defined is used will result in an empty string, there is not a simple way to error check this. 
The same will be true of the parallel PR to expand environment variables in the builder.conf used by the bundle-chroot-builder.py.

Reply:
I think it's feasible to check that env variables are actually set, with something (roughly) like:
 re := regexp.MustCompile("\\$\\{?[[:word:]]+\\}?")
    matches := re.FindAllString(i[m[1]:], -1)
    for _, s := range matches {
        if _, ok := os.LookupEnv(s[1:]); !ok {
            //print error & os.Exit()
....
If it's ok then *h.dest = os.ExpandEvn(i[m[1]:]) like you're doing already. I think it'd also be reasonable to say we should expect any env variable to just not be an empty string, since setting one to empty is valid and you can't tell the difference with ExpandEnv in that case. However if you say variables must be defined and not empty then it's easier just to check if ExpandEnv(s)  != "" and fail. Saves a later crash and/or content being generated in system locations they're not meant to go in.

Reply:
+1 screening with os.LookupEnv()

I think it's plausible someone might deliberately use an empty string (If it's the start of a path, for example, and not the entire value), though disallowing that seems fair enough. But at least checking that it's set seems both safer and relatively straightforward.

@tmarcu - Interesting, I didn't realize golang's regexp syntax accepted both \w and [[:word:]].

Reply:
@mdhorn  FYI -- you can do something similar in Python for your other PR with `if "FOO" in os.environ:`